test/libsolidity/semanticTests/libraries/library_return_struct.sol

```diff
@@ -0,0 +1,28 @@
+pragma experimental ABIEncoderV2;
+
+struct S {
+    uint a;
+    uint b;
+    uint c;
+}
+
+library lib {
+    function get(S storage self) public returns (S storage) {
+        return self;
+    }
+}
+
+contract TestContract {
+    using lib for S;
+    S s;
+
+    function f() public returns (uint) {
+        s.a = 0; s.b = 1; s.c = 2;
+        S storage ref = s.get();
```

Comment:
Even on this simple example this fails

Reply:
It looks to me as some problem with assigning to lvalue in `LValue::storeValue()`